ratpack-pac4j/src/test/groovy/ratpack/pac4j/internal/Pac4jAuthenticatorSpec.groovy

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ratpack.pac4j.internal
+
+import ratpack.handling.Context
+import ratpack.pac4j.RatpackPac4j
+import ratpack.path.PathBinding
+import ratpack.server.PublicAddress
+import ratpack.server.internal.ConstantPublicAddress
+import spock.lang.Specification
+import spock.lang.Unroll
+
+import static ratpack.pac4j.RatpackPac4j.DEFAULT_AUTHENTICATOR_PATH
+
+class Pac4jAuthenticatorSpec extends Specification {
```

Comment:
I don't think that this test is really necessary. I think that the convention is that we tend to use integration tests for everything unless there really is no other way and you have provided coverage for your changes by by unrolling the test in `OpenIdRpSpec`.